pkg/parquetquery/iters.go

```diff
@@ -80,17 +83,43 @@ func EqualRowNumber(upToDefinitionLevel int, a, b RowNumber) bool {
 		(a[2] == b[2] || upToDefinitionLevel < 2) &&
 		(a[3] == b[3] || upToDefinitionLevel < 3) &&
 		(a[4] == b[4] || upToDefinitionLevel < 4) &&
-		(a[5] == b[5] || upToDefinitionLevel < 5)
+		(a[5] == b[5] || upToDefinitionLevel < 5) &&
```

Comment:
i wonder if evaluating `upToDefinitionLevel < val` would be faster? i also wonder if this statement is too complex for the compiler to short circuit in the ways that make sense to us.

Reply:
i also thought about this, will follow up